Additional input options

flaviozavan opened the pull request:
Adds the possibility to configure disabling the input device while typing and
allows for choosing the click method. Both of which are useful for touchpads

----

Two points to consider:

1. I wasn't sure if the way I set the default values is within expectations or not;
2. I couldn't figure out how to run the tests.

inclement commented on config/config.toml:
```diff
@@ -241,6 +244,8 @@ scroll-button = 8
 middle-emulation = true
 left-handed = false
 natural-scroll = false
+disable-with-typing = true
```
Please name this `disable-while-typing` instead of `-with-`, to match the libinput name

flaviozavan replied:
I don't know where `with` came from. Changed.

inclement commented on config/viv_config.h:
```diff
@@ -128,6 +128,10 @@ struct viv_libinput_config the_libinput_configs[] = {
         .left_handed = 0,
         // Invert scroll direction
         .natural_scroll = 0,
+        // Disable while typing
```
Am I right to understand this is expected to be available only for touchpads? If so, could you tweak the comment to `Disable input while typing (usually touchpad only)` or similar?

flaviozavan replied:
done

inclement commented on config/viv_config.h:
```diff
@@ -128,6 +128,10 @@ struct viv_libinput_config the_libinput_configs[] = {
         .natural_scroll = 0,
+        // Disable while typing
+        .disable_with_typing = LIBINPUT_CONFIG_DWT_ENABLED,
+        // Method used for registering the type of click
```
Similarly, could you mention `(touchpad only)` here too?

flaviozavan replied:
done

inclement commented on src/viv_toml_config.c:
```diff
@@ -92,6 +92,13 @@ static struct string_map_pair scroll_method_map[] = {
 
+static struct string_map_pair click_method_map[] = {
+    {"none", LIBINPUT_CONFIG_CLICK_METHOD_NONE},
+    {"area", LIBINPUT_CONFIG_CLICK_METHOD_BUTTON_AREAS},
```
The tests have picked up a typo - looks like this should be `"areas"`?

flaviozavan replied:
excellent, fixed

inclement reviewed (changes requested):
Thanks very much, looks great except some minor tweaks.

The tests needed manually starting as that seems to be the default for a first PR from someone, but I think they should now run automatically on future commits.

inclement commented:
> I wasn't sure if the way I set the default values is within expectations or not;

I think ideally these should be pulled out to some centralised defaults list, but I hadn't thought about it before (and it didn't come up with the other libinput values) so there isn't an obvious one right now. Happy to merge as-is and change it later, I've made #92  to think about it.

That said, I think there's a case for making both options disabled by default - dwt in particular can be annoying for some (although essential for others, it's just subjective and hardware-dependent). Still happy to merge as-is, but I might push to change the defaults later.

flaviozavan commented:
Addressed the comments. About the defaults. I don't really care about the ones I chose, they were simply my touchpad's default and I wanted to change them. I made it default to none/disabled now. The centralized defaults idea seems like a pretty good idea.

I figured out how to run the tests locally this time. So they should pass now.

inclement commented:
Great, thanks again.
